event/handler_test.go

```diff
@@ -1,29 +1,62 @@
 package event_test
 
 import (
+	"context"
+	"errors"
 	"os"
 	"testing"
 
 	"github.com/ONSdigital/dp-search-data-extractor/config"
 	"github.com/ONSdigital/dp-search-data-extractor/event"
+	eventMock "github.com/ONSdigital/dp-search-data-extractor/event/mock"
 	. "github.com/smartystreets/goconvey/convey"
 )
 
-// TODO: remove hello called example test
-func TestHelloCalledHandler_Handle(t *testing.T) {
+var errZebedee = errors.New("zebedee test error")
+var getPublishDataFuncInError = func(ctx context.Context, uriString string) ([]byte, error) {
+	return nil, errZebedee
+}
+
+var getPublishDataFunc = func(ctx context.Context, uriString string) ([]byte, error) {
+	data := []byte("test data")
+	return data, nil
+}
+
+func TestContentPublishedHandler_Handle(t *testing.T) {
 
 	Convey("Given a successful event handler, when Handle is triggered", t, func() {
-		eventHandler := &event.HelloCalledHandler{}
-		filePath := "/tmp/helloworld.txt"
+		var zebedeeMock = &eventMock.ZebedeeClientMock{GetPublishedDataFunc: getPublishDataFunc}
+		eventHandler := &event.ContentPublishedHandler{zebedeeMock}
+		filePath := "/tmp/dpSearchDataExtractor.txt"
 		os.Remove(filePath)
 		err := eventHandler.Handle(testCtx, &config.Config{OutputFilePath: filePath}, &testEvent)
 		So(err, ShouldBeNil)
+		So(zebedeeMock.GetPublishedDataCalls(), ShouldNotBeEmpty)
+		So(zebedeeMock.GetPublishedDataCalls(), ShouldHaveLength, 1)
+		So(zebedeeMock.GetPublishedDataCalls()[0].UriString, ShouldEqual, testEvent.URL)
+	})
+
+	Convey("Given a un-successful event handler, when Handle is triggered", t, func() {
```

Comment:
Similarly with this one I would have this: 
```suggestion
	Convey("Given an event handler, not working successfully, and an event containing a URL", t, func() {
		var zebedeeMockInError = &eventMock.ZebedeeClientMock{GetPublishedDataFunc: getPublishDataFuncInError}
		eventHandler := &event.ContentPublishedHandler{zebedeeMockInError}
		filePath := ""

		Convey("When the event handler is triggered", func() {
			err := eventHandler.Handle(testCtx, &config.Config{OutputFilePath: filePath}, &testEvent)

			Convey("Then the published data endpoint should be called, using the URL, and an error returned", func() {
				So(err, ShouldNotBeNil)
				So(err.Error(), ShouldEqual, errZebedee.Error())
				So(zebedeeMockInError.GetPublishedDataCalls(), ShouldNotBeEmpty)
				So(zebedeeMockInError.GetPublishedDataCalls(), ShouldHaveLength, 1)
				So(zebedeeMockInError.GetPublishedDataCalls()[0].UriString, ShouldEqual, testEvent.URL)
			})
		})
	})
```